pennylane/devices/qubit/preprocess.py

```diff
@@ -249,13 +273,9 @@ def preprocess(
             options for the execution.
 
     Returns:
-        Tuple[Sequence[.QuantumTape], callable]: Returns a tuple containing
-        the sequence of circuits to be executed, and a post-processing function
-        to be applied to the list of evaluated circuit results.
+        Tuple[QuantumTape], Callable, ExecutionConfig: QuantumTapes that the device can natively execute,
+        a postprocessing function to be called after execution, and a configuration with unset specifications filled in.
     """
-    if execution_config.shots is not None:
-        # Finite shot support will be added later
-        raise DeviceError("The Python Device does not support finite shots.")
```

Comment:
Do we have already support for finite shots on the new device API?

Reply:
shots are now on the `QuantumScript`, so they've been removed from the execution config.

A different PR in the near future will add preprocessing for finite shot situations.

Reply:
Thank you for that!